### ISSUE TITLE
radically different results with Quantized version

### ISSUE THREAD

**Issue by dcsan** [facebookresearch/fastText#365]
Using quantization seems to give terrible results.

I am getting results from a .bin file and .ftz file

with quant:
![image](https://user-images.githubusercontent.com/514002/33132662-ddf60284-cf5f-11e7-9820-fa398bba17d3.png)

without
![image](https://user-images.githubusercontent.com/514002/33132550-903dd5da-cf5f-11e7-9bb9-f549365deea7.png)

(note, this is using some chinese data)

**Comment by dcsan** [comment]
for the quantize stage, it seems we can add all the extra parameters as for training, even though perhaps these are not needed?


./bin/fastText/fasttext supervised -input ./data/ignored/training/label-data.txt -output ./data/ignored/training/labels-model -pretrainedVectors wiki.zh.vec -dim 300 -loss hs -epoch 25 -minn 2 -wordNgrams 3

117 	 14:14:00 > DONE fullTrain

117 	 14:14:00 > START Quantizing...

./bin/fastText/fasttext quantize -output ./data/ignored/training/labels-model -input ./data/ignored/training/label-data.txt -pretrainedVectors /opt/rikai/common/data/ignored/wiki.zh.vec -retrain -dim 300 -loss hs -epoch 25 -minn 2 -wordNgrams 3
809 	 14:25:32 > DONE Quantizing

Also, training takes 2 minutes, and quantize a lot longer - 13 mins.  Is this expected?

I also tried without `-retrain` option

what other options could I try for this? Does quantizing work better with different values for epoch etc?

I am using the pretrained chinese word vectors, so had to set `-dim 300`

appreciate any suggestions!

**Comment by dcsan** [comment]
train
```

  cmd="$FT_BINARY supervised \
    -input ${LABEL_FILE} \
    -output ${MODEL_FILE} \
    -pretrainedVectors $WIKI_FILE \
    -dim 300 \
    -loss hs \
    -epoch 25 \
    -minn 2 \
    -wordNgrams 3"
  logit "train> $cmd"
  $cmd



  cmd="$FT_BINARY quantize \
    -output ${MODEL_FILE} \
    -input ${LABEL_FILE} \
    -pretrainedVectors $WIKI_FILE \
    -retrain \
    -dim 300 \
    -loss hs \
    -epoch 25 \
    -minn 2 \
    -wordNgrams 3"

   $cmd

```

**Comment by cpuhrsch** [comment]
Hello @dcsan,

Thank you for your post. Unfortunately we cannot help you with this very well, because we don't have access to the data and also because we can't offer general consulting on machine learning problems. Further, there has been no regression in performance on our end. 

Presumably you'll need to try more hyperparameters and track your performance across them. We've had [good experiences](https://fasttext.cc/docs/en/supervised-models.html) using "-qnorm -retrain -cutoff 100000" for quantize. You'll only need to use these parameters for "quantize" subcommand as the other parameters are retrieved from the original training settings. "quantize" might take longer than training, because quantizing the input matrix can take quite a bit of time.  I'm closing this issue now, but please feel encouraged to reopen it at any time.

Thanks,
Christian

**Comment by dcsan** [comment]
@cpuhrsch 

I'm seeing similar huge gap in performance for a german language project

The 'training data' is basically FB provided pretrainedVectors and a few items to be classified as below.

normal

![image](https://user-images.githubusercontent.com/514002/38414972-9debabc6-39c2-11e8-8ef0-6a0941779537.png)

quantized 

![image](https://user-images.githubusercontent.com/514002/38414941-8603e078-39c2-11e8-983d-e37f7c7bd590.png)


the memory usage is significantly better for the quant version - 1.5gb vs. 5gb

normal 5gb memory

![image](https://user-images.githubusercontent.com/514002/38415010-bcd223da-39c2-11e8-8868-4b7b4c13bd90.png)

quantized 1.5GB

![image](https://user-images.githubusercontent.com/514002/38415044-dd046c4e-39c2-11e8-8402-e5e8438ab905.png)


I'm not able to reopen the issue.

**Comment by dcsan** [comment]
failed to quant:

2	Fri Apr 6 17:57:42 CST 2018	 > START fullTrain -wordNgrams 2 -minn 2 -loss softmax -dim 300
1471	Fri Apr 6 18:22:11 CST 2018	 < DONE fullTrain
1471	Fri Apr 6 18:22:11 CST 2018	 > START quantize -qnorm -retrain -cutoff 200000
